apps/main/src/dex/components/PageRouterSwap/index.tsx

```diff
@@ -98,30 +97,54 @@ const QuickSwap = ({
 
   const isReady = pageLoaded && !isLoadingApi && isPageVisible
   const haveSigner = !!signerAddress
+
   const userFromBalance = userBalancesMapper[fromAddress]
   const userToBalance = userBalancesMapper[toAddress]
+
   const fromUsdRate = usdRatesMapper[fromAddress]
   const toUsdRate = usdRatesMapper[toAddress]
-  const fromToken = tokensNameMapper[fromAddress] ?? ''
-  const toToken = tokensNameMapper[toAddress] ?? ''
 
   const tokensMapperNonSmallTvlStr = useMemo(
     () => getTokensMapperStr(tokensMapperNonSmallTvl),
     [tokensMapperNonSmallTvl],
   )
 
-  const selectFromTokensList = useMemo(
-    () => getTokensObjList(selectFromList ?? selectToList, tokensMapper),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [selectFromList, selectToListStr, tokensMapperStr],
+  function getTokensObjList(tokensList: string[] | undefined, tokensMapper: TokensMapper | undefined) {
+    if (!tokensList || tokensList.length === 0 || !tokensMapper || Object.keys(tokensMapper).length === 0) return []
+    return tokensList.map((address) => tokensMapper[address])
+  }
+
+  const tokensFrom = useMemo(
+    () =>
+      getTokensObjList(selectFromList ?? selectToList, tokensMapper)
+        .filter((token) => !!token)
+        .map((token) => ({
+          chain: network?.networkId ?? '',
+          address: token?.address as `0x${string}`,
+          symbol: token?.symbol,
+          label: '',
+          volume: token?.volume ?? 0,
```

Comment:
Zero and empty strings should not be the fallback

```suggestion
          chain: network?.networkId,
          address: token?.address as Address,
          symbol: token?.symbol,
          volume: token?.volume,
```